src/kakarot/instructions/system_operations.cairo

```diff
@@ -548,6 +538,129 @@ namespace CallHelper {
+    //         via last twenty bytes of the keccak hash of:
+    //         keccak256(rlp([sender_address,sender_nonce]))[
+    //         See [CREATE](https://www.evm.codes/#f0)
+    // @param sender_address - The evm sender address.
+    // @param bytecode_len - The length of the initialization code.
+    // @param nonce - The nonce given to the create opcode.
+    // @return The pointer to the updated calling context.
+    func get_create_address{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(sender_address: felt, nonce: felt) -> (evm_contract_address: felt) {
+        // dummy impl to aide refactoring
+        return (0xAbdE100700000000000000000000000000000000 + nonce,);
+    }
+
+    // @notice Constructs an evm contract address for the create2 opcode
+    //         via last twenty bytes of the keccak hash of:
+    //         keccak256(0xff + sender_address + salt +
+    //         keccak256(initialisation_code))[12:]
+    //         See [CREATE2](https://www.evm.codes/#f5)
+    // @param sender_address - The evm sender address.
+    // @param bytecode_len - The length of the initialization code.
+    // @param bytecode - The offset to store the element at.
+    // @param salt - The salt given to the create2 opcode.
+    // @return The pointer to the updated calling context.
+    func get_create2_address{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(sender_address: felt, bytecode_len: felt, bytecode: felt*, salt: Uint256) -> (
+        evm_contract_address: felt
+    ) {
+        alloc_locals;
+        let (keccak_ptr: felt*) = alloc();
+        local keccak_ptr_start: felt* = keccak_ptr;
+
+        let (local bytecode_bytes8: felt*) = alloc();
+        Helpers.bytes_to_bytes8_little_endian(
+            bytes_len=bytecode_len,
+            bytes=bytecode,
+            index=0,
+            size=bytecode_len,
+            bytes8=0,
+            bytes8_shift=0,
+            dest=bytecode_bytes8,
+            dest_index=0,
+        );
+        with keccak_ptr {
+            // get keccak hash of bytecode
+            let (bytecode_hash_bigend) = keccak_bigend(
+                inputs=bytecode_bytes8, n_bytes=bytecode_len
+            );
+            // get keccak hash of
+            // marker + caller_address + salt + bytecode_hash
+            let (local packed_bytes: felt*) = alloc();
+
+            // 0xff is by convention the marker involved in deterministic address creation for create2
+            let (packed_bytes_len) = Helpers.felt_to_bytes(0xff, 0, packed_bytes);
+
+            let (sender_address_high, sender_address_low) = split_felt(sender_address);
+            let (
+                sender_address_bytes_array_len, sender_address_bytes_array
+            ) = Helpers.uint256_to_bytes_array(
```

Comment:
isn't there a `felt_to_bytes_array`?

Reply:
I initially was using that, but ran into an overflow error that came up when the testrunner ran this test: https://github.com/sayajin-labs/kakarot/blob/main/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py#L349

You can see the runtime error output here: https://github.com/sayajin-labs/kakarot/actions/runs/3955455088/jobs/6773922685#step:9:477

My interpretation was that because the address as encoded into a felt had overflow when dividing by 256: https://github.com/sayajin-labs/kakarot/blob/734606cf99777cb040bb8926fac4c5ada89bdc3f/src/utils/utils.cairo#L622

Let me know if my interpretation was wrong here.